src/write/buf_write.rs

```diff
@@ -0,0 +1,29 @@
+use std::{
+    io,
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+pub(crate) trait AsyncBufWrite {
+    /// Attempt to return an internal buffer to write to, flushing data out to the inner reader if
```

Comment:
Given the implementation that follows, I'd slightly reword this as follows:
```suggestion
    /// Attempt to return an internal buffer to write to, flushing pending `produce`d
    /// data out to the inner writer.
```